### PR TITLE
Add Additional Logging for new endpoint

### PR DESCRIPTION
**Pull request by tyler-spangler6**
## What was the problem?
This adds a new field to logging and updates an existing field for only the new expanded classifier.  

Previous Logging:
```json
{
    "vagov_claim_id": 100,
    "claim_type": "new",
    "classification_code": 8989,
    "classification_name": "Mental Disorders",
    "contention_text": "PTSD (post-traumatic stress disorder)",
    "diagnostic_code": "None",
    "is_in_dropdown": true,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/va-gov-claim-classifier",
    "date": "2024-11-20 14:20:24",
    "level": "info"
}
{
    "vagov_claim_id": 100,
    "claim_type": "claim_for_increase",
    "classification_code": 8940,
    "classification_name": "Cancer - Musculoskeletal - Other",
    "contention_text": "unmapped contention text",
    "diagnostic_code": 5012,
    "is_in_dropdown": false,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/va-gov-claim-classifier",
    "date": "2024-11-20 14:20:24",
    "level": "info"
}
{
    "vagov_claim_id": 100,
    "claim_type": "new",
    "classification_code": null,
    "classification_name": null,
    "contention_text": "unmapped contention text",
    "diagnostic_code": 8550,
    "is_in_dropdown": false,
    "is_lookup_table_match": false,
    "is_multi_contention": true,
    "endpoint": "/va-gov-claim-classifier",
    "date": "2024-11-20 14:20:24",
    "level": "info"
}
{
    "claim_id": 100,
    "form526_submission_id": 500,
    "is_fully_classified": false,
    "percent_clasified": 66.66666666666666,
    "num_processed_contentions": 3,
    "num_classified_contentions": 2,
    "endpoint": "/va-gov-claim-classifier",
    "date": "2024-11-20 14:20:24",
    "level": "info"
}
{
    "process_time": 0.004019021987915039,
    "url": "/va-gov-claim-classifier",
    "date": "2024-11-20 14:20:24",
    "level": "info"
}
```
New Logging for expanded classifier:
```json
{
    "vagov_claim_id": 100,
    "claim_type": "new",
    "classification_code": 8989,
    "classification_name": "Mental Disorders",
    "contention_text": "PTSD (post-traumatic stress disorder)",
    "processed_contention_text": "ptsd post traumatic stress disorder",
    "diagnostic_code": "None",
    "is_in_dropdown": true,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-20 14:27:09",
    "level": "info"
}
{
    "vagov_claim_id": 100,
    "claim_type": "claim_for_increase",
    "classification_code": 8940,
    "classification_name": "Cancer - Musculoskeletal - Other",
    "contention_text": "unmapped contention text ['']",
    "processed_contention_text": "",
    "diagnostic_code": 5012,
    "is_in_dropdown": false,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-20 14:27:09",
    "level": "info"
}
{
    "vagov_claim_id": 100,
    "claim_type": "new",
    "classification_code": 8997,
    "classification_name": "Musculoskeletal - Knee",
    "contention_text": "unmapped contention text ['acl tear']",
    "processed_contention_text": "acl tear",
    "diagnostic_code": 8550,
    "is_in_dropdown": false,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-20 14:27:09",
    "level": "info"
}
{
    "claim_id": 100,
    "form526_submission_id": 500,
    "is_fully_classified": true,
    "percent_clasified": 100.0,
    "num_processed_contentions": 3,
    "num_classified_contentions": 3,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-20 14:27:09",
    "level": "info"
}
{
    "process_time": 0.004132986068725586,
    "url": "/expanded-contention-classification",
    "date": "2024-11-20 14:27:09",
    "level": "info"
}
```
The additional field of `processed_classified_text` will us compare classifications based on the text processing lookup. This should also prevent logging any PII because we will only log the contention text if there is a match with the lookup table after processing.  Additionally adding the processed text in the contention text field will help with our DataDog widgets to differentiate between unclassifiable unmapped text and classifiable unmapped text.  

Associated tickets or Slack threads:
- https://github.com/orgs/department-of-veterans-affairs/projects/873/views/2?pane=issue&itemId=88018308&issue=department-of-veterans-affairs%7Cvagov-claim-classification%7C643

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: domain-cc/cc-app/src/python_src/api.py [hunk 1]
```diff
@@ -108,20 +108,40 @@ def log_contention_stats(
...
+                    "processed_contention_text": processed_text,
+                    "contention_text": log_contention_text,
+                }
+            )
+        # log none as the processed text if it is not in the LUT and leave unmapped contention text as is
+        else:
+            logging_dict.update({"processed_contention_text": None})
+        log_as_json(logging_dict)
+    else:
+        log_as_json(logging_dict)
```
Comment (williamphelps13) [inline domain-cc/cc-app/src/python_src/api.py]: What if broke this code out into a helper function? Something like:
```
def _update_logging_for_expanded_contention(
    logging_dict: dict, contention_text: str, log_contention_text: str
):
    """
    # Updates the logging dictionary for `/expanded-contention-classification`
    """
    processed_text = expanded_lookup_table.prep_incoming_text(contention_text)
    classification = expanded_lookup_table.get(contention_text)

    if classification.get("classification_code"):
        if log_contention_text == "unmapped contention text":
            log_contention_text = f"unmapped contention text {[processed_text]}"

        logging_dict.update(
            {
                "processed_contention_text": processed_text,
                "contention_text": log_contention_text,
            }
        )
    # log none as the processed text if it is not in the LUT and leave unmapped contention text as is
    else:
        logging_dict.update({"processed_contention_text": None})

    return logging_dict
```

That is then used like:
```
 if request.url.path == "/expanded-contention-classification":
        logging_dict = _update_logging_for_expanded_contention(
            logging_dict, contention_text, log_contention_text
        )

    log_as_json(logging_dict)
```

**Review by williamphelps13 — COMMENTED** [review]
Nice work here! Just one comment - Could the new code in `log_contention_stats` be broken into its own helper function?

**Review by williamphelps13 — APPROVED** [review]
Thanks for checking out the comment! Glad it looked like a positive change. This PR looks good to me.

**Review by raudabaugh — APPROVED** [review]

File: domain-cc/cc-app/src/python_src/api.py [hunk 2]
```diff
@@ -108,20 +132,27 @@ def log_contention_stats(
...
+        "endpoint": request.url.path,
+    }
+
+    if request.url.path == "/expanded-contention-classification":
+        logging_dict = log_expanded_contention_text(
+            logging_dict, contention.contention_text, log_contention_text
+        )
+
+        # log_as_json(logging_dict)
+    # else:
```
Comment (gabezurita) [inline domain-cc/cc-app/src/python_src/api.py:154]: Is this commented-out code left in deliberately?

**Review by gabezurita — APPROVED** [review]
LGTM!

File: domain-cc/cc-app/src/python_src/api.py (hunk 2 above)
Comment (tyler-spangler6) [reply domain-cc/cc-app/src/python_src/api.py:154]: @gabezurita Sorry I missed this comment before merging. That was accidentally left behind, I will clean this up in a following PR.

**Pull request merged**